core/metacling/src/TClingCallFunc.cxx

```diff
@@ -229,6 +229,30 @@ void TClingCallFunc::collect_type_info(QualType &QT, ostringstream &typedefbuf,
    GetTypeAsString(QT, type_name, C, Policy);
 }
 
+static bool IsCopyConstructorDeleted(QualType QT)
+{
+   CXXRecordDecl *RD = QT->getAsCXXRecordDecl();
+   if (!RD) {
+      // For types that are not C++ records (such as PODs), we assume that they are copyable, ie their copy constructor
+      // is not deleted.
+      return false;
+   }
+
+   RD = RD->getDefinition();
+   assert(RD && "expecting a definition");
+
+   if (RD->hasSimpleCopyConstructor())
+      return false;
+
+   for (auto *Ctor : RD->ctors()) {
+      if (Ctor->isCopyConstructor()) {
+         return Ctor->isDeleted();
```

Comment:
I would argue that no because it may be unexpected for the user that `TClingCallFunc` moves its objects when they "forget" to make their copy constructor `public`...

Reply:
it seems github or I did not publish the following comment/reply:

Well I am still confused.  It seems that we now get different behavior whether the user 'deletes' the copy constructor in pre-C++11 ways or post=C++1 ways:
```
class OldStyle {
   OldStyle(const OldStyle&); // Intentionally private and not implemented.
};
class NewStyle {
   NewStyle(const NewStyle&) = delete;
};
```
Making a copy constructor `private` is an explicit operation so I don't think consider the case '"forget" to make public' as a major concern.
So my question is why would the user would 'expects' when "TClingCallFunc to move its objects when the copy constructor is deleted" but not when the copy constructor is private.  What difference is semantic do you see?

Reply:
We cannot disambiguate "Intentionally private and not implemented" and forgotten, whereas `delete` is explicit. Users have to update their code to use C++11.

Reply:
> and forgotten

How can it be 'forgotten' since making it private requires intentional typing?

Reply:
Well we could check if there is a copy constructor that is private I think. The question is what's the 'right' behavior in this case...